### PR TITLE
Store adminSplit on work itself

### PR DESCRIPTION
**Pull request by harmvandendorpel**
Before this, all works shared one `adminSplit` percentage.

### CODE REVIEW

File: contracts/LeftGalleryController.sol [hunk 1]
```diff
@@ -126,6 +130,7 @@ contract LeftGalleryController is Ownable {
             works[workId].artist,
             works[workId].editions,
             works[workId].price,
+            works[workId].adminSplit,
```
Comment (harmvandendorpel) [inline contracts/LeftGalleryController.sol:133]: Would it make the contract 'cheaper' if we define an internal private method that emits the `updatedWork` event? And we call that instead each time the work is updated? I see some repetition.

File: contracts/LeftGalleryController.sol [hunk 2]
```diff
@@ -85,6 +86,7 @@ contract LeftGalleryController is Ownable {
         address payable artist,
         uint256 editions,
         uint256 price,
+        uint256 adminSplit,
         bool _paused
```
Comment (harmvandendorpel) [inline contracts/LeftGalleryController.sol:90]: why is this argument prefixed with underscore?

File: contracts/LeftGalleryController.sol [hunk 3]
```diff
@@ -198,9 +205,10 @@ contract LeftGalleryController is Ownable {
         );
     }
 
-    function updateAdminSplit(uint256 _adminSplit) public onlyOwner {
+    function updateAdminSplit(uint256 workId, uint256 _adminSplit) public onlyOwner {
```
Comment (harmvandendorpel) [inline contracts/LeftGalleryController.sol]: I guess `_adminSplit` can be renamed to `adminSplit` now.

File: contracts/LeftGalleryController.sol (hunk 1 above)
Comment (vrde) [reply contracts/LeftGalleryController.sol:133]: Would you mind elaborate on the internal private method?

**Comment by vrde** [comment]
Regarding events, we can save some gas by limiting the data emitted. We should emit the minimum amount of data to disambiguate an event, and we can always query the smart contract for the extra information we need that was not in the event itself.

For some events, emitting only the `workId` can suffice to retrieve the rest of the data.

**Pull request closed without merging**